### ISSUE TITLE
PieChart: outerRadius ignored when data contains multiple elements

### ISSUE THREAD

**Issue by hayzamjs** [techniq/layerchart#287]
When using the `PieChart` component from layerchart, the `outerRadius` property is correctly applied when the data array contains only one element. However, when the data array contains multiple elements, the outerRadius is ignored, and the chart defaults to a different radius. This behavior is inconsistent and impacts scenarios where dynamic radius adjustments are required for charts with multiple data points.

I have setup a playground instance [here](https://svelte.dev/playground/548a38c1d41b43e0b75b62d9dad1d903?version=5.2.7) to showcase the bug. If for some reason you're not able to access that this is the example code I used:

```svelte
<script>
	import { Legend, PieChart } from 'layerchart';

	let randomOuterRadius = $state(140);

	setInterval(() => {
		randomOuterRadius = Math.random() * (140 - 5) + 5;
	}, 1000);
</script>

<div class="custom-container">
	<h1>Outer Radius Test</h1>
	<div class="charts-container">
		<div class="chart-wrapper">
			<PieChart
				outerRadius={randomOuterRadius}
				innerRadius={70}
				data={[
					{ label: 'Single Value', count: 10 },
				]}
				key="label"
				value="count"
				padding={{ top: -10, right: 0, bottom: 20, left: 0 }}
				cRange={['#225DE5', '#12E0DD']}
			>
			</PieChart>
		</div>
		<div class="chart-wrapper">
			<PieChart
				outerRadius={randomOuterRadius}
				innerRadius={70}
				data={[
					{ label: 'First Value', count: 10 },
					{ label: 'Second Value', count: 20 },
				]}
				key="label"
				value="count"
				padding={{ top: -10, right: 0, bottom: 20, left: 0 }}
				cRange={['#225DE5', '#12E0DD']}
			>
			</PieChart>
		</div>
	</div>
</div>

<style>
h1 {
	color: black;
	text-align: center;
	margin-bottom: 1rem;
}

.custom-container {
	grid-column: span 12 / span 12;
	min-height: 410px;
	height: auto;
	background-color: white;
	border-radius: 12px;
	padding: 1rem;
}

.charts-container {
	display: flex;
	justify-content: space-between;
	align-items: center;
	gap: 1rem;
	width: 100%;
	height: 400px;
}

.chart-wrapper {
	width: 48%;
	height: 100%;
	position: relative;
	border-radius: 8px;
}
</style>
```

I don't know if I'm using the library incorrectly, but if this is indeed a bug, please let me know how I could help debug or provide more details.

**Comment by techniq** [comment]
Hi @hayzamjs, thanks for the report and reproduction.  I was also able to reproduce it in the docs and will need to investigate. 

PieChart currently has 2 [modes](https://github.com/techniq/layerchart/blob/2d743954c05c86fe198cf5ff1cf9f826574ce86f/packages/layerchart/src/lib/components/charts/PieChart.svelte#L168-L213), "single arc" and "pie" that is determined based on the series and data.

Currently the outer radius is used as an offset for multi-series, and since by default there is a single series, it is being [ignored](https://github.com/techniq/layerchart/blob/2d743954c05c86fe198cf5ff1cf9f826574ce86f/packages/layerchart/src/lib/components/charts/PieChart.svelte#L201), unlike when a single arc is [used](https://github.com/techniq/layerchart/blob/2d743954c05c86fe198cf5ff1cf9f826574ce86f/packages/layerchart/src/lib/components/charts/PieChart.svelte#L176).

I had spent a good bit of time to try to smartly switch between these modes, but need to think through this use case and possibly have a better signal (or just be explicit).

I'm working on heading out of town tomorrow so not likely to have a solution for a bit, especially if it takes a bit of investigation, but hopefully a simple solution reveals itself.  Worst case I'll have it fixed when I get back.

**Comment by techniq** [comment]
Fixed via [0.59.5](https://github.com/techniq/layerchart/releases/tag/layerchart%400.59.5).  Thanks again for the report

**Comment by hayzamjs** [comment]
Thanks so much for taking the time to reproduce the issue and for getting it fixed so quickly in 0.59.5! Really appreciate your effort on this. Safe travels!
